### ISSUE TITLE
Can not cancel mwc-tab-bar activated

### ISSUE THREAD

**Issue by govis** [material-components/material-components-web-components#804]
- [x] bug report
- [ ] feature request

<code>MDCTabBar:activated</code> event, even though dispatched with <code>cancelable: true</code> in <code>notifyTabActivated</code>, can not be cancelled  - the tab bar does not currently check if was cancelled.

Also need to know if it was activated by the user (by clicking a tab) versus a programmatic <code>activeIndex</code> change. Just like how checkbox and switch have <code>change</code> event that fires when user interacts with them and have <code>checked</code> property that can change independently.

**Comment by govis** [comment]
It appears that <code>MDCTab:interacted</code> event of the <code>mwc-tab</code> can be cancelled, so maybe it can be brought up to the tab bar level and <code>MDCTabBar:activated</code> can become activeIndex change or something to that extend (e.q. post interaction)

**Comment by govis** [comment]
Here's a repro/use case:

<pre><code>&lt;mwc-tab-bar @MDCTabBar:activated="${this._changePage}">
    &lt;mwc-tab id="page1">Page 1&lt;/mwc-tab>
    &lt;mwc-tab id="page2">Page 2&lt;/mwc-tab>
    &lt;mwc-tab id="page3">Page 2&lt;/mwc-tab>
&lt;/mwc-tab-bar>

_changePage(event) {
    let pageUri = getPageUri(event.detail); //tab that raised the event is not included in event.detail, need to jump through some hoops to get it
    //invoke page navigation using window.history.pushState/popstate
    this._navigate(pageUri);
    //cancel the event so that previously selected tab remains active
    event.preventDefault();
}

_navigate(pageUri) {
    //something like this...
    window.history.pushState({}, '', pageUri);
    window.dispatchEvent(new CustomEvent('location-changed'));
}</code></pre>

Basically we are using tabs for page navigation where pages have their own tab bars with appropriately pre-selected tab. When user clicks on another tab the app will navigate to the new page, but we need to cancel the tab selection as the page navigated from needs to keep the selected tab intact so that it matches the page.

Thank you

**Comment by asyncLiz** [comment]
Thanks for the use case! This type of UX isn't supported by MDC or MWC, I'd encourage you to extend `<mwc-tab-bar>` or `<mwc-tab>` if you want to add this functionality for your project.

Our recommended UX is for each page to have its own content, but share the same tabs. The content can be switched while the tabs remain constant and maintain their animation/UX when clicking on them.

**Comment by govis** [comment]
I am extending <code>mwc-tab-bar</code> and listening to <code>MDCTab:interacted</code> event raised by <code>mwc-tab</code> but it can't be cancelled either. It feels that design of wdc component is not well thought through. Fairly straight forward things such as getting the tab in mwc-tab-bar activate event are not possible #327
